muskrat/s3consumer.py

```diff
@@ -125,12 +142,8 @@ def consume(self):
         #Update: actually... this doesn't seem to be a problem...
         msg_iterator = self._get_msg_iterator()
 
-        for msg in msg_iterator:
-            #Sub 'directories' are prefix objects, so ignore them
-            if isinstance( msg, boto.s3.key.Key ):
-                self.callback( msg.get_contents_as_string() )
-                self._cursor.update( msg.name )
-
+        for obj in self._cursor.each(msg_iterator):
+            self.callback(obj.get()['Body'].read())
```

Comment:
Is satisfying this test really the behavior that we want though? Seems useful that you could write a consumer on root of the bucket, and it would process all entries. What are the downsides to including keys that go deeper than your prefix?